crates/rollup-boost/src/server.rs

```diff
@@ -125,7 +128,10 @@ impl RollupBoostServer {
             .await;
 
         // async call to builder to sync the builder node
-        if !self.execution_mode().is_disabled() {
+        if !self.execution_mode().is_disabled()
+            && (self.allow_traffic_to_unhealthy_builder
+                || matches!(self.probes.health(), Health::Healthy))
```

Comment:
minor nit but maybe extract the healthy thing into a variable this is a bit weird to read